### DISCUSSION TITLE
Instructions for use

### DISCUSSION THREAD

**Discussion by wahaha-1** [KumarRobotics/CoverageControl#4] [General]
Sorry to bother you again, I don't quite understand how to use your repo. Looking at your tutorial, I roughly understand that you use the python file in scripts and the configuration file in params to perform actions.
but
**1.** How to use lpac to guide ontology program?

**2.** How to simulate the actual map using the data from https://www.openstreetmap.org/ as in the paper?

**3.** How to call the downloaded gnuplot, ffmpeg plug-in to visualize and generate videos?

  Another question is that I only found the folder that saves the data under the lpac file. 
**4.** Is there a folder elsewhere in the project that saves the data during the running of the program?

**I hope there is a basic and complete guidance on the usage process.**
*(I have read your paper and am very interested in your ideas, especially those screenshots of experimental data. I thought I could understand how your program operates by reading the paper carefully. Sorry, I still overestimate my ability. , I have no choice but to bother you again)*

**Comment by wahaha-1** [comment]
This is the result after I use it now
![bfbf5657c9668f2648eff9395a34e4f0](https://github.com/KumarRobotics/CoverageControl/assets/153339362/5bb486bd-d811-4c16-8a27-e6bbedf70d58)

**Comment by AgarwalSaurav** [comment]
## LPAC Neural Network
For comprehensive guidelines on data generation, training, and evaluation of the LPAC neural network, please refer to our detailed documentation:
[LPAC Neural Network Documentation](https://kumarrobotics.github.io/CoverageControl/lpac.html).

To modify settings, please adjust the parameters in the TOML files as necessary.

For details on obtaining and using the model discussed in our publication, refer to the `Evaluation` section of the same documentation.

## Generating Videos
To facilitate video generation:
- Use the `RecordPlotData()` function to capture plot data at every time step, e.g., inside the `while` loop (line 52) in [coverage_class.py](https://github.com/KumarRobotics/CoverageControl/blob/main/python/scripts/coverage_env/coverage_class.py)
- Use the `RenderRecordedMap("dir/", "video.mp4")` function to render videos at the end of the program.

To activate video generation during the LPAC NN evaluation, set `GenerateVideo = true` in `eval_single.toml`. See an example TOML file here: [Example eval_single.toml](https://github.com/KumarRobotics/CoverageControl/blob/main/params/eval_single.toml).

Run the evaluation script `eval_single_env.py`, as described here: [Evaluation Scripts](https://kumarrobotics.github.io/CoverageControl/lpac.html#autotoc_md30).

## Documentation and API Access
Our reference manual provides comprehensive documentation for all available classes and functions:
[Reference Manual](https://kumarrobotics.github.io/CoverageControl/topics.html).

Please note that most functions available in the C++ Core API can also be accessed using Python.

---
The code for extracting data from OpenStreetMap has not yet been uploaded. Please feel free to ask any further questions or request additional information as needed.

**Comment by wahaha-1** [comment]
Thank you so much for your quick reply,！！！！
I'll try it soon！！！！
🫣🫣🫣🫣
I will be more detailed ，thank you very much for your patience~

**Comment by wahaha-1** [comment]
Under your guidance, the third and fourth problems have been successfully solved~~~expressing my gratitude to you again！！

But now there are other questions:
**1.** **How to customize the points of interest (areas with importance 1) and specify the initial position of the robot? I have not found a file location that can be adjusted for the time being; (I am particularly interested in it. If the data from the openmap website cannot be used, it is also very useful for me to design the map that needs to be covered and specify the initial position of the robot)**

**2.** Can I choose not to delete the generated motion trajectory pictures after generating the video?

**3.** I am curious about CoverageControl/python/scripts/training/train_cnn.py. Is it similar to train_lpac.py?

**4.** How to use the trained model file? Now I only know that the trained network will be used in the evaluation phase, but the models folder is not used. I don't quite understand this. In my understanding, training - generating model - calling model, I don't know how to reflect this last step.

**5.** There is a small error in the documentation
` python python/scripts/evaluators/eval_single_dataset.py \
${CoverageControl_ws}/lpac/params/eval_single_env.toml `
Should be changed to:
` python python/scripts/evaluators/eval_single_env.py \
${CoverageControl_ws}/lpac/params/eval_single.toml `

**Reply by AgarwalSaurav** [reply]
1. See `eval_single_env.py`. It loads the points of interest and initial positions from files.
2. Set `RobotPosHistorySize = 1` in `coverage_control_params.toml`
3. Yes, they are similar.
4. You can specify the model file in `eval_single.toml`
5. Thanks.

Try modifying the scripts according to your use.

**Reply by wahaha-1** [reply]
thank you !!!!!   I'll try them soon~~

**Comment by wahaha-1** [comment]
forgot one thing. After I used `RecordPlotData()`, I did not find the generated csv file. `RenderRecordedMap() `ran normally. I found four instructions about `RecordPlotData` in the document, but the calls failed.

**Comment by wahaha-1** [comment]
> forgot one thing. After I used `RecordPlotData()`, I did not find the generated csv file. `RenderRecordedMap() `ran normally. I found four instructions about `RecordPlotData` in the document, but the calls failed.


I understand this. The `RenderRecordedMap()`  is based on the data of `RecordPlotData(),` but it does not generate a `.csv` file like this [Evaluation.](https://kumarrobotics.github.io/CoverageControl/lpac.html#autotoc_md30)
